homeassistant/components/arduino/sensor.py

```diff
@@ -58,6 +78,13 @@ def name(self):
         """Get the name of the sensor."""
         return self._name
 
-    def update(self):
-        """Get the latest value from the pin."""
-        self._value = self._board.get_analog_inputs()[self._pin][1]
+    def cb_value_changed(self, data):
+        """Set the value from the data passed in the callback."""
+        self._value = data[DATA_VALUE]
+        self.async_write_ha_state()
```

Comment:
From what I understand the callbacks will be called from sync context by the library so we can't use the async api. Use `self.schedule_update_ha_state` instead.

Reply:
Can you point me to a page in the documentation, that explains the difference? Or another web page? It's not clear to me what you mean with 'from sync context'. Keep in mind that this is my first change to home assisstant and it's my first time coding in Python.

Reply:
Please read our asyncio docs. They explain this:
https://developers.home-assistant.io/docs/asyncio_index

Basically the asyncio api is not threadsafe by default so we have to take care to use the correct version of our api sync or async depending on where the call happens, outside the event loop or inside the event loop, ie the main thread.

Asyncio in Python is not considered easy to fully grasp immediately, unfortunately.

Reply:
The introduction video of asyncio in the documentation explains it very well, the concepts are similar to async/await in .NET. 

Now I understand why I should use the function you mentioned, thanks for the information! I will do the changes as soon as possible.